bokehjs/test/integration/regressions.ts

```diff
@@ -881,19 +881,32 @@ describe("Bug", () => {
   })
 
   describe("in issue #10498", () => {
-    it("prevents GridBox from rebuilding when rows or cols properties are modified", async () => {
-      const p1 = fig([300, 300])
-      const p2 = fig([300, 300])
-      p1.scatter({x: [0, 1], y: [0, 1], color: "red"})
-      p2.scatter({x: [1, 0], y: [0, 1], color: "green"})
-      const box = new GridBox({
-        children: [[p1, 0, 0], [p2, 0, 1]],
-        cols: ["300px", "300px"],
-        sizing_mode: "fixed",
-      })
-      const {view} = await display(box, [600, 300])
-      box.cols = ["100px", "500px"]
-      await view.ready
+    const params: Array<"cols" | "rows"> = ["cols", "rows"]
+    params.forEach(uses => {
+      it(
+        uses === "cols"
+          ? "prevents GridBox from rebuilding in x direction when cols are modified"
+          : "prevents GridBox from rebuilding in y direction when rows are modified",
+        async () => {
+          const p1 = fig([300, 300])
+          const p2 = fig([300, 300])
+          p1.scatter({x: [0, 1], y: [0, 1], color: "red"})
+          p2.scatter({x: [1, 0], y: [0, 1], color: "green"})
+          const box = new GridBox({
+            children: [
+              [p1, 0, 0],
+              uses === "cols" ?
+                [p2, 0, 1] :
+                [p2, 1, 0],
+            ],
+            [uses]: ["300px", "300px"],
+            sizing_mode: "fixed",
+          })
+          const {view} = await display(box, uses === "cols" ? [600, 300] : [300, 600])
+          box[uses] = ["100px", "500px"]
+          await view.ready
+        },
+      )
```

Comment:
Let's restore the original test and just add another case as you did in the original PR. This is really hard to read and a lot of complexity for I'm not sure what gain. A little duplication is perfectly fine. I know we have some tests that look similar to this, but I'm not too fond of them either.

Reply:
Sure! But how does it look now with your suggestions applied? Do you still find the near-duplicated version easier?

Reply:
The current state is definitely an improvement. I suppose this is fine for now. Let's update baselines and be done with this.